graph/pull.go

```diff
@@ -265,6 +265,10 @@ func (s *TagStore) pullImage(r *registry.Session, out io.Writer, imgID, endpoint
 	// FIXME: Try to stream the images?
 	// FIXME: Launch the getRemoteImage() in goroutines
 
+	layers := append(history, imgID)
+	s.graph.Retain(layers, imgID)
+	defer s.graph.Unretain(history, imgID)
+
```

Comment:
I don't see anywhere later on in `pullRepository` where we `Unretain` again after it's tagged. We need to make sure we add one there.